src/Parser/PotreeBinParser.js

```diff
@@ -1,53 +1,122 @@
         }
 
         const view = new DataView(buffer);
         // Format: X1,Y1,Z1,R1,G1,B1,A1,[...],XN,YN,ZN,RN,GN,BN,AN
-        const numPoints = Math.floor(buffer.byteLength / 16);
+        let pointByteSize = 0;
+        for (const potreeName of pointAttributes) {
+            const attr = POINT_ATTTRIBUTES[potreeName];
+            pointByteSize += attr.numElements * (attr.numByte || attr.arrayType.BYTES_PER_ELEMENT);
+        }
+        const numPoints = Math.floor(buffer.byteLength / pointByteSize);
 
-        const positions = new Float32Array(3 * numPoints);
-        const colors = new Uint8Array(4 * numPoints);
+        const attrs = [];
+        // get the variable attributes
+        for (const potreeName of pointAttributes) {
+            const attr = POINT_ATTTRIBUTES[potreeName];
+            attrs.push({
+                potreeName,
+                numElements: attr.numElements,
+                attributeName: attr.attributeName,
+                normalized: attr.normalized,
+                array: new attr.arrayType(attr.numElements * numPoints),
+                numByte: attr.numByte,
+            });
+        }
 
+        // init thightbbox
         const tightbbox = new THREE.Box3();
         tightbbox.min.set(Infinity, Infinity, Infinity);
         tightbbox.max.set(-Infinity, -Infinity, -Infinity);
         const tmp = new THREE.Vector3();
 
         let offset = 0;
-        for (let i = 0; i < numPoints; i++) {
-            positions[3 * i] = view.getUint32(offset + 0, true);
-            positions[3 * i + 1] = view.getUint32(offset + 4, true);
-            positions[3 * i + 2] = view.getUint32(offset + 8, true);
-
-            tmp.fromArray(positions, 3 * i);
-            tightbbox.min.min(tmp);
-            tightbbox.max.max(tmp);
+        for (let pntIdx = 0; pntIdx < numPoints; pntIdx++) {
+            for (const attr of attrs) {
+                for (let elemIdx = 0; elemIdx < attr.numElements; elemIdx++) {
+                    const fnName = `getUint${(attr.numByte || attr.array.BYTES_PER_ELEMENT) * 8}`;
+                    attr.array[pntIdx * attr.numElements + elemIdx] = view[fnName](offset, true);
 
-            colors[4 * i] = view.getUint8(offset + 12);
-            colors[4 * i + 1] = view.getUint8(offset + 13);
-            colors[4 * i + 2] = view.getUint8(offset + 14);
-            colors[4 * i + 3] = 255;
+                    offset += (attr.numByte || attr.array.BYTES_PER_ELEMENT);
+                }
 
-            offset += 16;
+                // calculate tightbox
+                if (attr.potreeName === 'POSITION_CARTESIAN') {
+                    tmp.set(attr.array[pntIdx * attr.numElements + 0],
+                            attr.array[pntIdx * attr.numElements + 1],
+                            attr.array[pntIdx * attr.numElements + 2]);
+                    tightbbox.min.min(tmp);
+                    tightbbox.max.max(tmp);
+                }
```

Comment:
would not it be simpler to use `geometry.computeBoundingBox()` ?

Reply:
Indeed. I've checked the implementation, and it is essentially the same as here. It'll add a for loop, but I don't expect its cost to be perceivable.